docs/socket-activation.rst

```diff
@@ -0,0 +1,47 @@
+Socket Activation
+-----------------
+
+While waitress does not support the various implementations of socket activation,
+for example using systemd or launchd, it is prepared to receive pre-bound applications
+from an init system.
+
+The following shows a code example starting waitress with three different
+pre-bound sockets.
+
+.. code-block:: python
+
+    import socket
+    import waitress
+
+
+    def app(environ, start_response):
+        content_length = environ.get('CONTENT_LENGTH', None)
+        if content_length is not None:
+            content_length = int(content_length)
+        body = environ['wsgi.input'].read(content_length)
+        content_length = str(len(body))
+        start_response(
+            '200 OK',
+            [('Content-Length', content_length), ('Content-Type', 'text/plain')]
+        )
+        return [body]
+
+
+    if __name__ == '__main__':
+        sockets = [
+            socket.socket(socket.AF_INET, socket.SOCK_STREAM),
+            socket.socket(socket.AF_INET, socket.SOCK_STREAM),
+            socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)]
+        sockets[0].bind(('127.0.0.1', 8080))
+        sockets[1].bind(('127.0.0.1', 9090))
+        sockets[2].bind('./test_socket')
```

Comment:
Mixing and matching TCP/IP and UNIX sockets is not something I'd recommend, I wonder if this is something we need to make sure is explicitly marked here.

Since UNIX Sockets can't provide the same information as TCP/IP sockets, waitress makes a lot of assumptions and fills in missing data. WSGI doesn't officially mention how to fill in the missing data so it is a complete guess.

Other settings, such as whether to enable proxy support are all or nothing for the waitress process, and exposing waitress on TCP/IP and on a unix socket you may want different settings for both.

Reply:
Hi @bertjwregeer, should I just mention that in the documentation or should I add a check for UNIX vs INET sockets that would throw an error in these cases?

I would probably lean towards adding the check, especially in light of the discussions for #207 and #171. The check can always be removed but it would be harder to add later on without breaking compatibility.

Let me know what you think.

Reply:
Add the check, and if people end up complaining because they really want to mix UNIX vs INET then we can revisit. Let's avoid giving people something that may leave them in a situation where their configuration leaves them wide open for potential security issues.